### ISSUE TITLE
ERROR Error: Uncaught (in promise): Error: StaticInjectorError[InjectionToken config]

### ISSUE THREAD

**Issue by digitalmedia34** [angular/angular#20513]
## I'm submitting a...

<pre><code>
[X ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
I have upgraded my current project from 4.4.4 to 5.0.0 using the recommandations from https://angular-update-guide.firebaseapp.com/

And now when I launch ng serve, the page keeps blank and the browser console returns the following error:

<pre><code>
ERROR Error: Uncaught (in promise): Error: StaticInjectorError[InjectionToken config]: 
  StaticInjectorError[InjectionToken config]: 
    NullInjectorError: No provider for InjectionToken config!
</code></pre>

I looked through my code and cant find reference to config. Also it occurs when navigating from one page to another using Router.

## Expected behavior
No error and navigate to new page.

## Minimal reproduction of the problem with instructions
I do not know how to reproduce this bug, my existing codebase is quite large.

I am aware it might be difficult to answer my issue, but I just ask to know if this error message rings a bell to anyone, so I know where to start digging.

## Environment

<pre><code>
Angular version: 5.0

"dependencies": {
    "@agm/core": "1.0.0-beta.2",
    "@angular/animations": "^5.0.0",
    "@angular/cdk": "5.0.0-rc0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/flex-layout": "2.0.0-beta.10",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/material": "5.0.0-rc0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/router": "^5.0.0",
    "@covalent/core": "1.0.0-rc.0",
    "@covalent/dynamic-forms": "1.0.0-rc.0",
    "@covalent/highlight": "1.0.0-rc.0",
    "@covalent/http": "1.0.0-rc.0",
    "@covalent/markdown": "1.0.0-rc.0",
    "@types/file-saver": "0.0.1",
    "angular-calendar": "^0.21.3",
    "angular-sortablejs": "^2.1.0",
    "core-js": "^2.5.1",
    "css-element-queries": "^0.4.0",
    "d3": "^3.5.17",
    "file-droppa": "^1.8.0",
    "file-saver": "^1.3.3",
    "hammerjs": "^2.0.8",
    "lodash": "^4.17.3",
    "moment": "^2.17.1",
    "ng2-currency-mask": "^4.3.1",
    "ng2-toastr": "4.1.2",
    "ng2-validation": "^4.2.0",
    "ngx-mask": "^2.1.11",
    "ngx-perfect-scrollbar": "^5.0.0-9",
    "ngx-quill": "^2.0.4",
    "normalize.css": "^5.0.0",
    "nvd3": "^1.8.5",
    "quill": "^1.3.4",
    "rxjs": "^5.5.2",
    "screenfull": "^3.0.2",
    "signalr": "^2.2.1",
    "smooth-scrollbar": "^8.2.0",
    "sortablejs": "^1.6.0",
    "ts-collections": "0.0.16",
    "ts-helpers": "^1.1.1",
    "underscore": "^1.8.3",
    "zone.js": "^0.8.14"
  },
  "devDependencies": {
    "@angular/cli": "latest",
    "@angular/compiler-cli": "^5.0.0",
    "@types/d3": "^3.5.17",
    "@types/googlemaps": "^3.26.14",
    "@types/jasmine": "2.5.53",
    "@types/lodash": "^4.14.44",
    "@types/node": "~6.0.60",
    "@types/nvd3": "^1.8.33",
    "@types/underscore": "^1.7.33",
    "codelyzer": "~3.2.0",
    "jasmine-core": "~2.6.2",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.2.0",
    "tslint": "~5.7.0",
    "typescript": "~2.4.2"
  }

Browser:
- [ X] Chrome (desktop) version 62
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  Windows 

Others:

</code></pre>

**Comment by ericmartinezr** [comment]
https://github.com/angular/angular/search?q=StaticInjectorError&type=Issues&utf8=✓

**Comment by jmac11** [comment]
Try and run it with chromium  instead of chrome browser with plunker is
better for your solution i believe.

**Comment by digitalmedia34** [comment]
Hi jmac not sure what chromium would do differently but I just tried it and it gave me the same error.

**Comment by jackjoy** [comment]
```Are you using lazy loading modules?Are you using npm link  for some packages?Similar issues for me 2 days ago after upgrade to @angular v5+node v8+npm v5,and i digged into found that webpack  duplicate some modules in separate bundles,which cause many StaticInjector errors.I am not sure the root cause is NPM or webpack,but seems not related to @angular.```

**Comment by digitalmedia34** [comment]
Hi @jackjoy how did you track down the duplication of the modules? Also if it were a npm or webpack issue how come it works with prev version of angular?

**Comment by jackjoy** [comment]
```First my project worked well with @angular@5.0, then i upgrade from nodejs@6.10+NPM@3.10 to nodejs@8.9+NPM@5.5,after reinstall dependencies, @angular was upgrade to v5.01, in dev mode, StaticInjector errors occurred in browser.But production mode with aot is OK.First i thought it was angular's issue.But after investigation, i found one dependency lib package use folder reference, e.g, "xxxx" : file:///... , not the normal package version.This cause webpack to duplicate some modules in lazy loading bundles from node_modules of 'xxxx' lib project folder, not from the app project's node_modules folder.Finally i changed the reference to normal format with package version, all works fine now.I am not sure the root cause and don't know why webpack has such strange behaviors.```

**Comment by digitalmedia34** [comment]
@jackjoy, thanks I am going to take a look and let you know

**Comment by nivida** [comment]
@digitalmedia34 Have you anything new to your issue? I'm actually struggling with the same problem.

**Comment by caseyPav** [comment]
I am also having this same issue- I've been chasing it on and off for days... debugging this is painful and has not yielded any more detailed info than the original error message... hopefully this is resolved or capable of returning a more meaningful error in the next update. Or someone kind points out how we're collectively doing something stupid : ) and it's a simple one line fix

**Comment by digitalmedia34** [comment]
@nivida, I haven't found a solution.

**Comment by nivida** [comment]
@digitalmedia34 after testing with many angular versions and firebase versions i have actually an package combination that works: 

```json
{
  "name": "backend",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "serve": "ng serve --preserve-link"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^5.0.0",
    "@angular/cdk": "^5.0.0-rc0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/material": "^5.0.0-rc0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/router": "^5.0.0",
    "angularfire2": "^5.0.0-rc.3",
    "core-js": "^2.4.1",
    "firebase": "^4.6.0",
    "rxjs": "^5.5.2",
    "zone.js": "^0.8.14"
  },
  "devDependencies": {
    "@angular/cli": "1.5.0",
    "@angular/compiler-cli": "^5.0.0",
    "@angular/language-service": "^5.0.0",
    "@types/jasmine": "~2.5.53",
    "@types/jasminewd2": "~2.0.2",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.2.0",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.2.0",
    "tslint": "~5.7.0",
    "typescript": "~2.4.2"
  }
}
```

**Comment by digitalmedia34** [comment]
Wow that's great @nivida, let me give it a try.

**Comment by caseyPav** [comment]
@digitalmedia34 your initial issue was caused by ngx-mask. 

checked our common libraries and set some debug points in the creation of their InjectorTokens, ngx-mask is 100% the culprit.

 It's removal results in a perfectly functional app for me. I suspect you were just loading NgxMaskModule (without the forRoot()) as I was as well... adding the for root let me hit the creation of the MaskService, where no matter how it's provided is always undefined. It will now give a similar stacktrace with it blowing up trying to map undefined now and pointing to the **source** of the issues finally...

 If you could confirm this by replicating on your end and checking value of config when hit I would appreciate it... would like to see if it's reproducible or unique to me. 
![image](https://user-images.githubusercontent.com/20408469/33223047-4a9b7a92-d12a-11e7-88bf-2d4670a1018c.png)

There is an existent thread[ here](https://github.com/JsDaddy/ngx-mask/issues/49) regarding this that I will update with my findings... I will be checking for a functional replacement or submitting a pull request with the necessary fixes. 

@nivida  I would suggest that if you're still getting the same error confirm there aren't any modules you're omitting a forRoot() from- adding that will likely generate a more meaningful error. @jackjoy was onto something with the pathing issue (config is imported from ./config)

**Comment by digitalmedia34** [comment]
Hi @caseyPav you are right, the culprit was NgxMaskModule. To solved i upgraded to 2.1.17 and added forRoot() to the module import.
